### PR TITLE
WT-10352 Implement delete_object() in connection class for Azure

### PR DESCRIPTION
**Pull request by mariammojid**
Implemented delete_object for Azure storage source using the Azure SDK function [DeleteIfExists](https://learn.microsoft.com/en-us/dotnet/api/azure.storage.blobs.specialized.blobbaseclient.deleteifexists?view=azure-dotnet#azure-storage-blobs-specialized-blobbaseclient-deleteifexists(azure-storage-blobs-models-deletesnapshotsoption-azure-storage-blobs-models-blobrequestconditions-system-threading-cancellationtoken))()

Assumptions: 
- object exists in a valid bucket

Dependencies:
- put_object implementation - combined with version in progress atm and delete passed unit testing
- object_exists to check for valid input
- bucket_exists

### CODE REVIEW

**Review by korteland — APPROVED** [review]
LGTM! One question, does the scope of this ticket include adding tests, or does that come later?

File: ext/storage_sources/azure_store/azure_connection.cpp [hunk 1]
```diff
@@ -64,9 +64,19 @@ azure_connection::put_object(const std::string &file_name) const
...
-azure_connection::delete_object() const
+azure_connection::delete_object(const std::string &object_name) const
 {
+    std::string obj = _object_prefix + object_name;
+
+    auto object_client = _azure_client.GetBlobClient(obj);
+    auto delete_blob_response = object_client.DeleteIfExists();
+
+    // Returns false if obj doesn't exist.
+    if (!delete_blob_response.Value.Deleted)
```
Comment (jiechenbo) [inline ext/storage_sources/azure_store/azure_connection.cpp:77]: add a message if the object doesn't exist to std::cerr

Comment (mariammojid) [reply ext/storage_sources/azure_store/azure_connection.cpp]: done :)

File: ext/storage_sources/azure_store/azure_connection.cpp [hunk 2]
```diff
@@ -64,9 +64,21 @@ azure_connection::put_object(const std::string &file_name) const
...
+azure_connection::delete_object(const std::string &object_key) const
 {
+    std::string obj = _object_prefix + object_key;
+
+    auto object_client = _azure_client.GetBlobClient(obj);
+    auto delete_blob_response = object_client.DeleteIfExists();
+
+    // Returns false if obj doesn't exist.
+    if (!delete_blob_response.Value.Deleted) {
+        std::cerr << obj + " : No such Object." << std::endl;
```
Comment (jiechenbo) [inline ext/storage_sources/azure_store/azure_connection.cpp:78]: np: I would place the comment as " : No such object file exists in the bucket."

**Review by jiechenbo — COMMENTED** [review]
Please add unit testing now for deleteObject functionality

**Comment by mariammojid** [comment]
evergreen merge

File: ext/storage_sources/azure_store/azure_connection.cpp (hunk 2 above)
Comment (mariammojid) [reply ext/storage_sources/azure_store/azure_connection.cpp]: done :)

**Comment by mariammojid** [comment]
evergreen merge

**Review by jiechenbo — APPROVED** [review]
lgtm

**Pull request merged**